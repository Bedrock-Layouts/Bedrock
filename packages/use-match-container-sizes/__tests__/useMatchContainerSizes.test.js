import useContainerQuery from "@bedrock-layout/use-container-query";
import React from "react";
import ReactDOM from "react-dom/";
import { act } from "react-dom/test-utils";

import useMatchContainerSizes from "../src";

jest.mock("@bedrock-layout/use-container-query");

useContainerQuery.mockImplementation(() => true);

let matches;
const HookWrapper = () => {
  const node = document.createElement("div");
  matches = useMatchContainerSizes(node);

  return null;
};

describe("useMatchContainerSizes", () => {
  let container;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });
  test("useMatchContainerSizes is not null", () => {
    expect(useMatchContainerSizes).toBeTruthy();
  });

  it("should call useContainerQuery", () => {
    act(() => {
      ReactDOM.render(<HookWrapper />, container);
    });

    expect(useContainerQuery).toBeCalled();

    expect(matches).toMatchSnapshot();
    expect(useContainerQuery.mock.calls).toMatchSnapshot();
  });
});
