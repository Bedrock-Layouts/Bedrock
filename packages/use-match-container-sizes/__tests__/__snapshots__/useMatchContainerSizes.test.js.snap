// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useMatchContainerSizes should call useContainerQuery 1`] = `
Object {
  "large": true,
  "largeOnly": true,
  "medium": true,
  "mediumOnly": true,
  "smallOnly": true,
  "xlarge": true,
  "xlargeOnly": true,
  "xxlarge": true,
}
`;

exports[`useMatchContainerSizes should call useContainerQuery 2`] = `
Array [
  Array [
    <div />,
    639,
    undefined,
  ],
  Array [
    <div />,
    640,
    1023,
  ],
  Array [
    <div />,
    1024,
    1199,
  ],
  Array [
    <div />,
    1200,
    1439,
  ],
  Array [
    <div />,
    1023,
    undefined,
  ],
  Array [
    <div />,
    1199,
    undefined,
  ],
  Array [
    <div />,
    1439,
    undefined,
  ],
  Array [
    <div />,
    1920,
    undefined,
  ],
]
`;
